### PR TITLE
Adds some high-level documentation to the contracts folder.

### PR DESCRIPTION
**Pull request by scottynomad**

### CODE REVIEW

File: contracts/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ERC1155 token representing the NRTs (carbon removal credits) issued to a supplier.
+
+Each token ID encodes the wallet address of the supplier to whom the NRTs were issued in addition to information about the source and location of the carbon removed. See [RemovalLib](./RemovalIdLib.sol) for details.
+
+The total supply of each Removal token ID represent the estimated quantity of NRTs (tonnes of CO2 removed from the atmosphere) on a specific parcel of land in a specific year.
+
+Lifecycle of a Removal:
+
+1. Minted to a supplier's address
+2. Ownership transfered to the _Market_ contract to be listed for sale.
```
Comment (jaycenhorton) [inline contracts/README.md:41]:
```suggestion
2. Ownership transferred to the _Market_ contract to be listed for sale.
```

File: contracts/README.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+### Certificate (NCCR)
+
+[source](./Certificate.sol)
+
+ERC721a token representing a carbon removal buyer's proof of purchase.
+
+Every purchase in the market mints a new _Certificate_ token with a unique ID. _Certificate_ is configured as an ERC1155 recipient to receive the purchased _Removal_ tokens.
+
+Certificates are non-transferrable. It is a fundamental tenet of the Nori market design that Removals and their Certificates cannot change hands after the purchase transaction.
```
Comment (jaycenhorton) [inline contracts/README.md:53]:
```suggestion
Certificates are non-transferable. It is a fundamental tenet of the Nori market design that Removals and their Certificates cannot change hands after the purchase transaction.
```

File: contracts/README.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+The logic for multiple cliffs followed by linear unlock used by the _LockedNORIV2_ contract. The vesting behavior and the unlocking behavior of each grant are each implemented using a schedule from this library.
+
+### RestrictedNORILib
+
+The schedule logic used in _RestrictedNORI_
+
+### RemovalQueue
+
+The queueing mechanism used by the _Market_ contract to maintain an ordered list of _Removal_ tokens listed for sale on behalf of a given supplier.
```
Comment (jaycenhorton) [inline contracts/README.md:105]:
```suggestion
The queuing mechanism used by the _Market_ contract to maintain an ordered list of _Removal_ tokens listed for sale on behalf of a given supplier.
```

**Review by jaycenhorton — APPROVED** [review]

File: contracts/README.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+Each token ID encodes the wallet address of the supplier to whom the NRTs were issued in addition to information about the source and location of the carbon removed. See [RemovalLib](./RemovalIdLib.sol) for details.
+
+The total supply of each Removal token ID represent the estimated quantity of NRTs (tonnes of CO2 removed from the atmosphere) on a specific parcel of land in a specific year.
+
+Lifecycle of a Removal:
+
+1. Minted to a supplier's address
+2. Ownership transfered to the _Market_ contract to be listed for sale.
+3. Sold by the _Market_ and ownership transferred to one or more _Certificate_
```
Comment (amiecorso) [inline contracts/README.md:42]: technically the ownership of the removal is transferred to the Certificate contract in all cases.  the accounting about which certificate IDs "own" the removal exists only in the certificate contract itself. not sure if that's worth being more specific about here

File: contracts/README.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+## Vesting and Lockup
+
+### LockedNORIV2
+
+[source](./LockedNORIV2.sol)
+
+Investor and Employee timed vesting and unlocking contract.
+
+Previously audited [here](). The current version was updated to support the new ERC20Permit pattern with _BridgedPolygonNORI_ and to remove reliance on the ERC777 tokensReceived callback.
```
Comment (amiecorso) [inline contracts/README.md:81]: still need a link here I think

File: contracts/README.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+[LockedNORI](https://polygonscan.com/token/0xccfffa6c2a030821331cc113b63babdc60bff82a) on PolygonScan.
+
+### RestrictedNORI
+
+Supplier insurance holdback contract.
+
+Operates similarly to _LockedNORI_ but simplified to remove cliffs and stacked vesting + unlock. Adds the ability to create more than one schedule per owner address and transferability of full or partial restricted balances between addresses.
```
Comment (amiecorso) [inline contracts/README.md:89]:
```suggestion
Operates similarly to _LockedNORI_ by acting as a wrapper token that governs the scheduled release of the underlying _BridgedPolygonNori_ asset.  Implemented as an ERC1155, each token id has its own schedule parameters that control the linear release of the underlying assets over the duration of the schedule.  It is possible to create more than one schedule per owner address and also to transfer full or partial restricted balances between addresses.
```

**Review by amiecorso — APPROVED** [review]

File: contracts/README.md (hunk 4 above)
Comment (scottynomad) [reply contracts/README.md:42]: Good point, I was simplifying that but it makes it unclear.

**Comment by scottynomad** [comment]
@jaycenhorton your spelling skill is admirable!

**Comment by jaycenhorton** [comment]
> @jaycenhorton your spelling skill is admirable!

can't take the credit-- cspell called out this one. I've considered adding [this](https://github.com/marketplace/actions/cspell-action) github action for those who don't use it locally.

**Pull request merged**